File: chapter-10/factorial.go
package main

import "fmt"

func factorial(i int) int {
	if i == 1 {
		return 1
	}
	return i * factorial(i - 1)
}

func main() {
	fmt.Println(factorial(5))
	fmt.Println(factorial(7))
	fmt.Println(factorial(9))
	fmt.Println(factorial(11))
}

File: chapter-11/factorial.py
def factorial(index):
  if index <= 1:
    return 1
  return index * factorial(index - 1)

def factorial_bottom(number, index = 1, product = 1):
  if index > number:
    return product
  return factorial_bottom(number, index + 1, product * index)

print(factorial(5))
print(factorial_bottom(5))
print(factorial(7))
print(factorial_bottom(7))
